doc/man3/EVP_PKEY_CTX_new.pod

```diff
@@ -52,23 +53,44 @@ If I<ctx> is NULL, nothing is done.
 
 =head1 NOTES
 
-=over 4
-
-=item 1.
+=head2 On B<EVP_PKEY_CTX>
 
 The B<EVP_PKEY_CTX> structure is an opaque public key algorithm context used
 by the OpenSSL high level public key API. Contexts B<MUST NOT> be shared between
 threads: that is it is not permissible to use the same context simultaneously
 in two threads.
 
-=item 2.
+=head2 On Key Types
 
 We mention "key type" in this manual, which is the same
 as "algorithm" in most cases, allowing either term to be used
 interchangeably.  There are algorithms where the I<key type> and the
 I<algorithm> of the operations that use the keys are not the same,
 such as EC keys being used for ECDSA and ECDH operations.
 
+Key types are given in two different manners:
+
+=over 4
+
+=item Legacy NID or EVP_PKEY type
+
+This is the I<id> used with EVP_PKEY_CTX_new_id().
+
+These are B<EVP_PKEY_RSA>, B<EVP_PKEY_RSA_PSS>, B<EVP_PKEY_DSA>,
+B<EVP_PKEY_DH>, B<EVP_PKEY_EC>, B<EVP_PKEY_SM2>, B<EVP_PKEY_X25519>,
+B<EVP_PKEY_X448>, and are used by legacy methods.
```

Comment:
I'm wondering since when NIDs are legacy, and does that mean the functions making use of an NID are deprecated?

Reply:
We have treated them as legacy for a long time now.  If you look at the `crypto/` code and search for `legacy:`, you'll find that it's all about old code that use NIDs as identifiers, while the provider based code do not use NIDs at all.

Note, however, that it's perfectly OK to use the constant NIDs we have _internally_.  We do that all over the place in our providers, quite visibly so in `providers/implementations/signature/`.  On the other hand, no NID is knowingly passing the libcrypto\<-\>provider interface, because that's a public interface.